src/gsi/hdraobmod.f90

```diff
@@ -0,0 +1,1315 @@
+  real(r_kind)                          ,intent(in   ) :: twindin
+  real(r_kind),dimension(nlat,nlon,nsig),intent(in   ) :: prsl_full,hgtl_full
+
+! Declare local parameters
+  real(r_kind),parameter:: r1_2 = 1.2_r_kind
+  real(r_kind),parameter:: r50  = 50.0_r_kind
+  real(r_kind),parameter:: r90  = 90.0_r_kind
+  real(r_kind),parameter:: r360 = 360.0_r_kind
+
+!  integer(i_kind),parameter:: mxtb=5000000
+!  integer(i_kind),parameter:: nmsgmax=100000 ! max message count
+  integer(i_kind), parameter:: maxlevs=20000
+
+! Declare local variables
+  logical tob,qob,uvob,psob
+  logical outside
+  logical,allocatable,dimension(:,:):: lmsg           ! set true when convinfo entry id found in a message
+
+  character(80) hdstr,hdstr2,hdstr3,levstr,hdtypestr
+  character(80) obstr
+  character(10) date
+  character(8) subset
+  character(8) prvstr,sprvstr     
+  character(8) c_station_id,id_station
+  character(8) stnid
+  character(8) stntype
+
+  integer(i_kind) ireadmg,ireadsb
+  integer(i_kind) lunin,i,maxobs,j,idomsfc,nmsgmax,mxtb
+  integer(i_kind) kk,klon1,klat1,klonp1,klatp1
+  integer(i_kind) nc,nx,ntread,itx,ii,ncsave
+  integer(i_kind) ihh,idd,idate,iret,im,iy,k,levs
+  integer(i_kind) kx,nreal,nchanl,ilat,ilon
+  integer(i_kind) lim_qm
+  integer(i_kind) iout
+  integer(i_kind) irec
+  integer(i_kind) ntest,nvtest
+  integer(i_kind) kl,k1,k2,k1_ps,k1_q,k1_t,k1_uv,k2_q,k2_t,k2_uv,k2_ps
+  integer(i_kind) itypex,id
+  integer(i_kind) minobs,minan
+  integer(i_kind) ntb,ntmatch,ncx,cat
+  integer(i_kind) nmsg                ! message index
+  integer(i_kind) jj,start,next,ncount_ps,ncount_q,ncount_uv,ncount_t
+  integer(i_kind),dimension(5):: idate5
+  integer(i_kind),dimension(nconvtype)::ntxall
+  integer(i_kind),dimension(nconvtype+1)::ntx
+  integer(i_kind),allocatable,dimension(:,:):: tab
+  integer(i_kind) igroup,istation
+  integer(i_kind) ierr_ps,ierr_q,ierr_t,ierr_uv   !  the position of error table collum
+  integer(i_kind),dimension(maxlevs):: pqm,qqm,tqm,wqm
+  integer(i_kind) nstations
+  integer(i_kind),allocatable,dimension(:):: list_stations
+  real(r_kind) time,timeobs,toff,t4dv,zeps
+  real(r_kind) qtflg,ediff,usage,ediff_ps,ediff_q,ediff_t,ediff_uv
+  real(r_kind) u0,v0,uob,vob,dx,dy,dx1,dy1,w00,w10,w01,w11
+  real(r_kind) qoe,qobcon,dlnpob,ppb,poe,obheight
+  real(r_kind) toe,woe,errout,oelev,dlat,dlon,dlat_earth,dlon_earth
+  real(r_kind) dlat_earth_deg,dlon_earth_deg
+  real(r_kind) elev,stnelev
+  real(r_kind) cdist,disterr,disterrmax,rlon00,rlat00
+  real(r_kind) vdisterrmax,u00,v00
+  real(r_kind) rminobs,rminan,es,dummy
+  real(r_kind) del,terrmin,werrmin,perrmin,qerrmin,del_ps,del_q,del_t,del_uv
+  real(r_kind) tsavg,ff10,sfcr,zz,deltat
+  real(r_kind) time_correction,fact
+  real(r_kind),dimension(nsig):: presl,hgtl,lnpresl
+  real(r_kind),dimension(nsig-1):: dpres
+  real(r_kind),dimension(maxlevs)::plevs
+  real(r_kind),allocatable,dimension(:,:):: cdata_all,cdata_out
+  real(r_kind) :: zob,tref
```

Comment:
Mike,

Sorry there were so many issues.  I will try to do better next time.

I made changes to address all of your issues (I hope).  You noted that twindin and ctwind were not used.  This indicated that I had not included a check to make sure it was inside the time window.  When I added that some obs were rejected.  This was because for some observation the reported seconds in the input bufr data is sometimes missing.  I put in a check for this and if missing use 0.  This changed the values in output a bit (because the time of the observation changed for the better).

I also made a change in format statements in statsconv.f90 and statslight.f90 because the number of observations was exceeding what the format statement allowed.  I6 to I7.